utils/databake/src/converter.rs

```diff
@@ -0,0 +1,261 @@
+///
+/// #[derive(Bake)]
+/// #[databake(path = my_crate)]
+/// struct Data {
+///     number: usize,
+///     string: AsStaticStr<String>, // can be written as StringAsStaticStr
+/// }
+///
+/// let data = Data {
+///     number: 6,
+///     string: 6.to_string().into()
+/// };
+///
+/// assert_eq!(
+///     data.bake(&Default::default()).to_string(),
+///     r#"my_crate :: Data { number : 6usize , string : "6" , }"#
+/// );
+///
+/// mod my_crate {
+///     struct Data {
+///         number: usize,
+///         string: &'static str,
+///     }
+/// }
+/// ```
+#[repr(transparent)]
+pub struct AsStaticStr<T>(pub T);
+
+impl<T> AsStaticStr<T> {
+    #[inline]
+    pub fn new(value: T) -> Self {
+        value.into()
+    }
+
+    #[inline]
+    pub fn into(self) -> T {
+        self.0
+    }
+}
+
+impl<T> Bake for AsStaticStr<T>
+where
+    T: AsRef<str>,
+{
+    fn bake(&self, _ctx: &CrateEnv) -> TokenStream {
+        let value = &self.0.as_ref();
+        quote!(#value)
+    }
+}
+
+impl<T> BakeSize for AsStaticStr<T>
+where
+    T: AsRef<str>,
+{
+    fn borrows_size(&self) -> usize {
+        self.0.as_ref().len()
+    }
+}
+
+impl<T> Default for AsStaticStr<T>
+where
+    T: Default,
+{
+    #[inline]
+    fn default() -> Self {
+        Self(T::default())
+    }
+}
+
+impl<T> Deref for AsStaticStr<T> {
+    type Target = T;
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T> DerefMut for AsStaticStr<T> {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl<T> From<T> for AsStaticStr<T> {
+    #[inline]
+    fn from(value: T) -> Self {
+        Self(value)
+    }
+}
+
+#[test]
+fn as_static_str() {
+    use std::borrow::Cow;
+
+    assert_eq!(
+        AsStaticStr(Cow::<str>::Owned("hi".to_string()))
+            .bake(&Default::default())
+            .to_string(),
+        r#""hi""#
+    );
+}
+
+/// Alias for [`AsStaticStr<String>`].
+pub type StringAsStaticStr = AsStaticStr<String>;
+
+/// Let [`Bake`] output a `&'static [T]` for anything that can be iterated over.
+///
+/// It can be helpful if the type needs to be constructed (e.g. [`Vec<T>`]) but should result in a `&'static [T]`.
+///
+/// This requires that the crate using the generated data needs a different struct.
+///
+/// ```
+/// use databake::{converter::AsRefSlice, Bake};
+///
+/// #[derive(Bake, Default)]
+/// #[databake(path = my_crate)]
+/// struct Data {
+///     numbers: AsRefSlice<Vec<usize>, usize>, // can be written as `VecAsRefSlice<usize>`
+/// }
+///
+/// let mut data = Data::default();
+/// data.numbers.push(6);
+///
+/// assert_eq!(
+///     data.bake(&Default::default()).to_string(),
+///     r#"my_crate :: Data { numbers : & [6usize ,] , }"#
+/// );
+///
+/// mod my_crate {
+///     struct Data {
+///         numbers: &'static [usize],
+///     }
+/// }
+/// ```
+#[repr(transparent)]
+pub struct AsRefSlice<B, T>(pub B, pub PhantomData<T>);
```

Comment:
This should be `IteratorIntoSlice`

Reply:
Valid point - was AsRef<[T]> before, but that does not work as well.

Maybe it's even enough when VecAsRefSlice is kept and AsRefSlice/IteratorIntoSlice is removed? Is such a generic case helpful?

Should it be IteratorIntoSlice or IteratorAsRefSlice (I've currently used the latter since the naming scheme it is identical to the other ones)

Reply:
The same goes for AsStaticStr, maybe only StringAsStaticStr is enough. I tried to be helpful with the generics, but I don't know how helpful it is.